compiler/src/test/java/io/neow3j/compiler/HashTest.java

```diff
@@ -54,20 +58,92 @@ public void hash256ReturnTypeShouldAppearAsHash256InDebugInfo() throws IOExcepti
                         "/HashTest$HashTestContract.java"))));
 
         assertThat(unit.getDebugInfo().getMethods().get(1).getName(),
-                is("io.neow3j.compiler.HashTest$HashTestContract,methodReturningHash256"));
+                is("io.neow3j.compiler.HashTest$HashTestContract,hash256FromBytes"));
         assertThat(unit.getDebugInfo().getMethods().get(1).getReturnType(),
                 is(ContractParameterType.HASH256.jsonValue()));
     }
 
+    @Test
+    public void testHash160FromString() throws IOException {
+        CompilationUnit compUnit = new Compiler().compile(HashFromStringContract.class.getName());
+        NeoMethod neoMethod = compUnit.getNeoModule().getSortedMethods().get(0);
+        assertThat(neoMethod.getInstructions().get(0).getOpcode(), is(OpCode.PUSHDATA1));
+        assertArrayEquals(neoMethod.getInstructions().get(0).getOperand(),
+                hexStringToByteArray(reverseHexString("0xcc5e4edd9f5f8dba8bb65734541df7a1c081c67b")));
+        assertThat(neoMethod.getInstructions().get(22).getOpcode(), is(OpCode.RET));
+    }
+
+    @Test
+    public void testHash256FromString() throws IOException {
+        CompilationUnit compUnit = new Compiler().compile(HashFromStringContract.class.getName());
+        NeoMethod neoMethod = compUnit.getNeoModule().getSortedMethods().get(1);
+        assertThat(neoMethod.getInstructions().get(0).getOpcode(), is(OpCode.PUSHDATA1));
+        assertArrayEquals(neoMethod.getInstructions().get(0).getOperand(),
+                hexStringToByteArray(
+                        reverseHexString("0xcb30ea3c29e205e8b1233ac7fa7fa51284c40ab920a915353376014871752ca4")));
+        assertThat(neoMethod.getInstructions().get(34).getOpcode(), is(OpCode.RET));
+    }
+
+    @Test
+    public void testInvalidHash160FromString() {
+        IllegalArgumentException thrown = assertThrows(IllegalArgumentException.class,
+                () -> new Compiler().compile(InvalidHash160.class.getName()));
+        assertThat(thrown.getMessage(), is("Hash must be 20 bytes long but was 19 bytes."));
+    }
+
+    @Test
+    public void testInvalidHash256FromString() {
+        IllegalArgumentException thrown = assertThrows(IllegalArgumentException.class,
+                () -> new Compiler().compile(InvalidHash256.class.getName()));
+        assertThat(thrown.getMessage(), is("Hash must be 32 bytes long but was 27 bytes."));
+    }
+
+    @Test
+    public void testHashFromStringWithNonConstantString() {
```

Comment:
There should be a test just like this for ECPoint in `ECPointTest`.

Reply:
This basically tests the method `ObjectsConverter.checkForConstantStringArgument()` which is called when handling `Hash160`, `Hash256`, or `ECPoint` instances with a string argument. So, do you think I should add a test for all three anyways?

Reply:
If we were testing the `checkForConstantStringArgument` directly one test would be enough. But here we're testing on a higher level. I think that it's necessary to test it for all three classes because we might change the behavior of specifically one in the future. Then a specific test will tell us what the expected behavior is/was.

Reply:
Good point, I'll add tests for all three.